R/bootstrap.R

```diff
@@ -63,24 +65,46 @@ bootstrapPage <- function(..., title = NULL, responsive = NULL, theme = NULL) {
 #' @export
 #' @keywords internal
 bootstrapLib <- function(theme = NULL) {
-  # Eventually this should be deprecated in favor of bootstraplib::bootstrap()
-  #shinyDeprecated("bootstraplib::bootstrap()", old = "shiny::bootstrapLib()")
-  if (!useBsTheme()) {
-    return(bootstrapDependency(theme))
-  }
-
-  if (!is.null(theme)) {
-    warning(
-      "The `theme` argument in `fluidPage()`, `bootstrapPage()`, etc. is not compatible ",
-      "with bootstraplib theming, so the bootstrap theme is being ignored in favor of ",
-      "theme='", theme, "'. To instead use the bootstraplib theme, remove the ",
-      "`theme` argument and use the bootstraplib equivalent (if you want a shinytheme, ",
-      "provide the theme name to bs_theme_new()'s `bootswatch` argument)."
-    )
-    return(bootstrapDependency(theme))
-  }
+  tagFunction(function() {
+    # If a bootstraplib theme isn't requested, return static/default Bootstrap
+    # build; otherwise, we're going be dynamically generating a new HTML
+    # dependency
+    if (!is_bs_theme(theme)) {
+      return(bootstrapDependency(theme))
+    }
```

Comment:
We might need another early exit here if bootstraplib is installed, but us a insufficient version